### DISCUSSION TITLE
Segment cleaning via automation

### DISCUSSION THREAD

**Discussion by csidirop** [Hypfer/Valetudo#2135] [Q&A / Support] [answered]
### Are you already using Valetudo?

Yes, for a long time

### Your questions

Hi, I want to setup a segment cleaning automation via Home Assistant (clean segment A x-times). This is fairly simple goal, but I'm stuck somehow. The segments are not exposed by mqtt and I can't find the REST API documentation anymore.

What would you suggest? And can you help me with the documentation, I just cant find it on the homepage?

**Comment by vivia** [comment, marked as answer]
Set up said segment cleaning automation on the Valetudo web interface and then long click the "Clean N segments" button. It will tell you the MQTT payload to call from Home Assistant.

The documentation for segment handling in MQTT is here: https://valetudo.cloud/pages/integrations/mqtt.html#segmentcleaningmapsegmentationcapability

**Reply by csidirop** [reply]
Great thank you! That was all I needed.
